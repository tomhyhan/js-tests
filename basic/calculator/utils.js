function add(x,y) {
    return x+y
}

function sub(x,y) {
    return x-y
}

function mul(x,y) {
    return x*y
}

function div(x,y) {
    if (y == 0) {
        throw new Error("Can not divide by 0")
    }
    return x / y
}

module.exports = {add, sub, div, mul}