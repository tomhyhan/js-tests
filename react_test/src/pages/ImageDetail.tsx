import React from 'react'
import { useLocation } from 'react-router-dom'

export default function ImageDetail() {
    const {
        state: {image},
    } = useLocation()
    
  return (
    <ul>
        <li>{image.id}</li>
        <li>{image.download_url}</li>
        <li>{image.author}</li>
    </ul>
  )
}
