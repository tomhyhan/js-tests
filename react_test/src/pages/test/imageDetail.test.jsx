import { render } from "@testing-library/react";
import ImageDetail from '../ImageDetail';
import { describe, it } from "vitest";
import { withRouter }from  "../../test/utils";
import { Route } from "react-router-dom";
import { fakeImage } from '../../test/fakeData';

describe("ImageDetail", () => {
    it("renders ImageDetail correctly with video state", async () => {
        console.log(fakeImage)
        const {asFragment} = render(withRouter(<Route path="/" element={<ImageDetail />} />,{
            pathname: "/",
            state: {image: fakeImage},
            key: "my-key"
        }))
        expect(asFragment()).toMatchSnapshot()
    })
})