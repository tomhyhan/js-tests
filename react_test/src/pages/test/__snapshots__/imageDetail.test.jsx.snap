// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageDetail > renders ImageDetail correctly with video state 1`] = `
<DocumentFragment>
  <ul>
    <li>
      2
    </li>
    <li>
      https://foo.bar.download_url/
    </li>
    <li>
      Bar
    </li>
  </ul>
</DocumentFragment>
`;
